library/src/main/java/com/novoda/downloadmanager/MigrationJob.java

```diff
@@ -103,10 +103,14 @@ private void migrateV1DataToV2Database(DownloadsPersistence downloadsPersistence
         downloadsPersistence.persistBatch(persistedBatch);
 
         for (Migration.FileMetadata fileMetadata : migration.getFileMetadata()) {
-            String url = fileMetadata.uri();
+            String url = fileMetadata.originalNetworkAddress();
 
+            FilePath filePath = new LiteFilePath(fileMetadata.originalFileLocation());
+            if (filePath.path() == null || filePath.path().isEmpty()) {
```

Comment:
If there is no existing path then we create it from the new persistence base path and the file name of whatever we are trying to download.

Reply:
This still suffers from the issue of duplicate files which will be sorted when we add a directory per batch.

Reply:
It might be worth adding this as an `Issue` so we won't forget